File: android/src/main/kotlin/com/bugreport/LogCategory.kt
package com.bugreport

/**
 * Log categories matching the JS SDK exactly.
 */
enum class LogCategory(val value: String) {
    CRASH("CRASH"),
    NETWORK("NETWORK"),
    RENDER("RENDER"),
    LIFECYCLE("LIFECYCLE"),
    PERF("PERF"),
    STORAGE("STORAGE"),
    AUDIO("AUDIO"),
    VIDEO("VIDEO"),
    EVAL("EVAL"),
    APP("APP"),
    USER("USER"),
    SYSTEM("SYSTEM");

    companion object {
        fun from(value: String) = entries.firstOrNull { it.value == value } ?: APP
    }
}

File: ios/Sources/BugReport/LogCategory.swift
// Log categories matching the JS & Kotlin SDKs exactly.

public enum LogCategory: String, Codable, CaseIterable {
    case CRASH
    case NETWORK
    case RENDER
    case LIFECYCLE
    case PERF
    case STORAGE
    case AUDIO
    case VIDEO
    case EVAL
    case APP
    case USER
    case SYSTEM

    public static func from(_ value: String) -> LogCategory {
        return LogCategory(rawValue: value) ?? .APP
    }
}
